[Python] string formatting concept exercise (#3034)
exercism/v3: languages/python/exercises/concept/pretty-leaflet/.docs/introduction.md

## string-formatting

> The Python `str` build-in type can be initialized using two robust string formatting methods. The string interpolation with `f'{variable}'` should be preferred as it is very readable, complete, and very fast. When a more versatile approach is needed, `''.format()` allows creating almost all the `str` you need.

Reviewer: ```suggestion
The Python `str` built-in can be initialized using two robust string formatting methods :  `f-strings` and `str.format()`. String interpolation with `f'{variable}'` is preferred because it is easily read, complete in place, and very fast. When an internationalization-friendly or more flexible approach is needed, `str.format()` allows creating almost all the other `str` variations you might  need.
```

# literal string interpolation. f-string

Literal string interpolation is a way of quickly and efficiently formatting and evaluating expressions to `str` using the `f` prefix and the curly brace `{object}`. It can be used with all enclosing string types as: single quote `'`, double quote `"` and for multi-lines and escaping triple quotes `'''` or `"""`.

In this basic example of **f-string**, the variable `name` is rendered at the beginning of the string, and the variable `age` of type `int` is converted to `str` and rendered after `' is '`.

```python
>>> name, age = 'Artemis', 21

> >>> f'{name} is {age} year old.'

Reviewer: ```suggestion
>>> f'{name} is {age} years old.'
```

> 'Artemis is 21 year old.'

Reviewer: ```suggestion
'Artemis is 21 years old.'
```

```

> The expressions evaluated can be almost anything, to mention some of the possibilities that can be evaluated: `str`, numbers, variables, arithmetic expressions, conditional expressions, built-in types, slices, functions or any objects with either `__str__` or `__repr__` methods defined. Some examples:

Reviewer: ```suggestion
The expressions evaluated by an `f-string` can be almost anything -- so the usual cautions about sanitizing input apply.  Some of the many possibilities that can be evaluated: `str`, numbers, variables, arithmetic expressions, conditional expressions, built-in types, slices, functions or any objects with either `__str__` or `__repr__` methods defined. Some examples:
```

```python
>>> waves = {'water': 1, 'light': 3, 'sound': 5}

>>> f'"A dict can be represented with f-string: {waves}."'
'"A dict can be represented with f-string: {\'water\': 1, \'light\': 3, \'sound\': 5}."'

>>> f'Tenfold the value of "light" is {waves["light"]*10}.'
'Tenfold the value of "light" is 30.'
```

> f-string supports control mechanisms such as width, alignment, precision that are described for `.format()`.

Reviewer: ```suggestion
f-string output supports the sam control mechanisms such as _width_, _alignment_, and _precision_ that are described for `.format()`.
```

# str.format() method

> The `str.format()` allow to replace placeholders with values, the placeholders are identified with named indexes {price} or numbered indexes {0} or empty placeholders {}. The values are specified as parameters in the `format` method. Example:

Reviewer: ```suggestion
`str.format()` allows for the  replacement of in-text placeholders.  Placeholders are identified with named indexes {price} or numbered indexes {0} or empty placeholders {}. Their values are specified as parameters in the `str.format()` method. Example:
```
